src/App.jsx

```diff
@@ -56,16 +60,39 @@ function App() {
       ? restaurants
       : restaurants.filter((r) => r.category === category);
 
+  // const handleClickItem = () => {
+  //   setModalOpen(true);
+  // };
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+  };
+
+  const handleClickItem = (restaurant) => {
+    setSelectedRestaurant(restaurant);
+    setModalOpen(true);
+  };
+
   return (
     <>
       <Header />
       <main>
         <CategoryFilter category={category} onChangeCategory={setCategory} />
-        <RestaurantList restaurants={filteredRestaurants} />
+        <RestaurantList
+          restaurants={filteredRestaurants}
+          onItemClick={handleClickItem}
+        />
       </main>
       <aside>
-        <RestaurantDetailModal />
-        <AddRestaurantModal />
+        {modalOpen && selectedRestaurant && (
+          <RestaurantDetailModal
+            restaurant={selectedRestaurant}
+            onClose={handleCloseModal}
+          />
+        )}
+        {addModalOpen && (
+          <AddRestaurantModal onClose={() => setAddModalOpen(false)} />
```

Comment:
```suggestion
          <AddRestaurantModal onClose={handleCloseModal} />
```
위에서 모달 닫는 함수를 만들어서 그걸 사용해도 좋을 것 같은데 주렁의 생각은 어떠신가요??🙃

Reply:
handleCloseModal를 지우고, 모달 닫기를 세부화하였습니다!
```
  const handleAddModalClose = () => setAddModalOpen(false);
  const handleDetailModalClose = () => setDetailModalOpen(false);
```
 이런식으로 만들고 
`<AddRestaurantModal onClose={handleAddModalClose} />`
이렇게 사용해볼게용

Reply:
```ts
const handleModalClose = (setModalOpenState) => {
  setModalOpenState(false);
}

<>
  <AddRestaurantModal onClose={() => handleModalClose(setAddModalOpen)} />
  <RestaurantDetailModal onClose={() => handleModalClose(setDetailModalOpen)} />
</>
```
이렇게 setState 함수를 인자로 넘기면 `handleModalClose` 하나로도 여러 모달을 끌 수 있을 것 같아요~!